### DISCUSSION TITLE
Can the rule compiler code be accessed when using Pocketbase as a framework?

### DISCUSSION THREAD

**Discussion by martinklepsch** [pocketbase/pocketbase#4487] [Q&A] [answered]
I'm looking at [these docs for extending with JS](https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state) and was thinking that it would be nice to specify access rules the same semi-plaintext way as its done for collections. I'm sure the backend somehow converts this into a datastructure, can this be accessed from the JSVM environment? 

Perhaps with support for reaching into query params? 

EDIT; I should also add; Thank you for Pocketbase, I'm really enjoying working with it!

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure if that is what you are looking for but you could try with [Check if record can be accessed](https://pocketbase.io/docs/js-records/#check-if-record-can-be-accessed).

Additionally, there are 2 query methods for record(s) retrieval currently exposed that operates with the same filter/rule syntax:

```js
// retrieve a single "articles" collection record by a string filter expression
// (use "{:placeholder}" to safely bind untrusted user input parameters)
const record = $app.dao().findFirstRecordByFilter(
    "articles", "status = 'public' && category = {:category}",
    { category: "news" },
)

// retrieve multiple "articles" collection records by a string filter expression
// (use "{:placeholder}" to safely bind untrusted user input parameters)
const records = $app.dao().findRecordsByFilter(
    "articles",                                    // collection
    "status = 'public' && category = {:category}", // filter
    "-publised",                                   // sort
    10,                                            // limit
    0,                                             // offset
    { category: "news" },                          // optional filter params
)
```

There is no mechanism at the moment for accessing the internals how a rule expression is parsed, if that is what you are looking for (_even in Go you can't intercept the parsing of the fields without manually creating a custom resolver; this may change with the refactoring but I haven't thought much about it yet_).

**Reply by martinklepsch** [reply]
Thanks for your answer, after learning more about how to use Pocketbase, I realized I was asking the wrong question. 😇
